embedchain/vectordb/pineconedb.py

```diff
@@ -0,0 +1,174 @@
+                    "Please make sure the type is right and that you are passing an instance."
+                )
+            self.config = config
+        self.client = self._setup_pinecone_index()
+        # Call parent init here because embedder is needed
+        super().__init__(config=self.config)
+
+    def _initialize(self):
+        """
+        This method is needed because `embedder` attribute needs to be set externally before it can be initialized.
+        """
+        if not self.embedder:
+            raise ValueError("Embedder not set. Please set an embedder with `set_embedder` before initialization.")
+
+    # Loads the Pinecone index or creates it if not present.
+    def _setup_pinecone_index(self):
+        pinecone.init(
+            api_key=os.environ.get("PINECONE_API_KEY"),
+            environment=os.environ.get("PINECONE_ENV"),
+        )
+        self.index_name = self._get_index_name()
+        indexes = pinecone.list_indexes()
+        if indexes is None or self.index_name not in indexes:
+            pinecone.create_index(
+                name=self.index_name,
+                metric=self.config.metric,
+                dimension=self.config.dimension
+            )
+        return pinecone.Index(self.index_name)
+
+    def get(
+            self, ids: Optional[List[str]] = None, where: Optional[Dict[str, any]] = None, limit: Optional[int] = None
+    ):
+        """
+        Get existing doc ids present in vector database
+
+        :param ids: _list of doc ids to check for existance
+        :type ids: List[str]
+        :param where: to filter data
+        :type where: Dict[str, any]
+        :return: ids
+        :rtype: Set[str]
+        """
+        zero_vector = []
+        for i in range(self.config.dimension):
+            zero_vector.append(0)
+
+        results = self.client.query(vector=zero_vector, top_k=max(1, self.count()), filter=where)
+        ids = set()
+
+        for result in results["matches"]:
+            ids.add(result["id"])
+        return {"ids": ids}
+
+    def add(self, documents: List[str], metadatas: List[object], ids: List[str]):
+        """add data in vector database
+
+        :param documents: list of texts to add
+        :type documents: List[str]
+        :param metadatas: list of metadata associated with docs
+        :type metadatas: List[object]
+        :param ids: ids of docs
+        :type ids: List[str]
+        """
+
+        docs = []
+        embeddings = self.embedder.embedding_fn(documents)
+        for id, text, metadata, embedding in zip(ids, documents, metadatas, embeddings):
+            metadata["text"] = text
+            docs.append(
+                {
+                    "id": id,
+                    "values": embedding,
+                    "metadata": metadata,
+                }
+            )
+        self.client.upsert(docs)
+
+    def query(self, input_query: List[str], n_results: int, where: Dict[str, any]) -> List[str]:
```

Comment:
@rupeshbansal i am not sure why we are expecting List[str] for input_query param, in reality it's always a string. this cause issue when querying, any input_text will return the same vectors

Reply:
Hey, completely agree with the observation, but somehow that is the abstraction that is being used for the query method of all the vector dbs in embedchain. I have opened a new issue (https://github.com/embedchain/embedchain/issues/776) to get this fixed